File: Meeting 11.12.21 - Set and Map/basicprogramming2.java
import java.util.*;
import java.io.*;

class basicprogramming2{
    public static ArrayList<Integer> arr;
    public static FastScanner sc;
    public static int n, t;

    public static void main(String[] args){
        sc = new FastScanner();
        n = sc.nextInt();
        t = sc.nextInt();
        arr = new ArrayList<>();
        for(int i = 0; i < n; i++){
            arr.add(sc.nextInt());
        }
        switch (t) {
            case 1: 
                twoSum(7777);
                break;
            case 2:
                unique();
                break;
            case 3: 
                findMode(n/2);
                break;
            case 4: 
                findMedian();
                break;
            case 5: 
                printRange(100, 999);
                break;
        }
    }


    private static void printRange(int min, int max) {
        Collections.sort(arr);
        StringBuilder sb = new StringBuilder();
        for(int val : arr){
            if(val < min) continue;
            if(val > max) break;
            sb.append(val).append(' ');
        }
        System.out.println(sb);
    }


    private static void findMedian() {
        Collections.sort(arr);
        if(n%2 == 1){
            System.out.println(arr.get(n/2));
            return;
        }
        System.out.println(arr.get(n/2 - 1) + " " + arr.get(n/2));
    }

    private static void findMode(int targetQuantity) {
        int num = -1;
        TreeMap<Integer, Integer> m = new TreeMap<>();
        for(int i : arr){
            if(m.containsKey(i)){
                m.replace(i, m.get(i)+1);
            } else {
                m.put(i, 1);
            }
        }
        for(Map.Entry<Integer, Integer> entry : m.entrySet()){
            int val = entry.getValue();
            if(val > targetQuantity){
                num = entry.getKey();
            }
        }
        System.out.println(num);
    }

    private static void unique() {
        TreeSet<Integer> s = new TreeSet<>();
        for(int i : arr) {
            if(!s.add(i)) {
                System.out.println("Contains duplicate");
                return;
            }
            // s.add(i);
        }
        System.out.println("Unique");
    }

    private static void twoSum(int target) {
        TreeSet<Integer> s = new TreeSet<>();
        for(int i : arr) {
            if(s.contains(target - i)){
                System.out.println("Yes");
                return;
            }
            s.add(i);
        }
        System.out.println("No");
    }

    static class FastScanner {
        BufferedReader br=new BufferedReader(new InputStreamReader(System.in));
        StringTokenizer st=new StringTokenizer("");
        String next() {
            while (!st.hasMoreTokens())
                try {
                    st=new StringTokenizer(br.readLine());
                } catch (IOException e) {
                    e.printStackTrace();
                }
            return st.nextToken();
        }

        int nextInt() {
            return Integer.parseInt(next());
        }
        int[] readArray(int n) {
            int[] a=new int[n];
            for (int i=0; i<n; i++) a[i]=nextInt();
            return a;
        }
        long nextLong() {
            return Long.parseLong(next());
        }
        double nextDouble() {
            return Double.parseDouble(next());
        }
        String nextLine() {
            try {
                return br.readLine();
            } catch(IOException e) {
                throw new RuntimeException(e);
            }
        }
    }
}

File: Meeting 11.12.21 - Set and Map/basicprogramming2.py
from sys import stdin 

def action1(n, arr):
    set_num = set()
    for i in arr:
        if 7777 - i in set_num:
            print("Yes")
            return
        else:
            set_num.add(i)
    print("No")

"""
arr = [1 7770 3 4 5 6 7]

i = 1
s.count(7776 [7777 - 1]) == 0
s.insert(1) // set s = {1}

i = 7770
s.count(7 [7777 - 7770]) == 0
s.insert(7770) // set s = {1, 7770}

...

// set = {1, 7770, 3, 4, 5, 6}
s = 7 
s.count(7770 [7777 - 7]) == 1
--> print("Yes")

"""


def action2(arr):
    set_num = set()
    for i in arr:
        if i in set_num:
            print("Contains duplicate")
            return
        set_num.add(i)
    print("Unique")



def action3(n, arr):
    dict_num = dict()
    for i in arr:
        if i in dict_num:
            dict_num[i] = dict_num[i] + 1
        else:
            dict_num[i] = 1
        if dict_num[i] > n / 2:
            print(i)
            return
    print(-1)

    
def action4(n, arr):
    arr.sort()
    if len(arr) % 2 == 0:
        print(arr[n // 2 - 1], arr[n // 2])
    else:
        print(arr[(n - 1) // 2])


def action5(arr):
    arr.sort()
    for i in arr:
        if i < 100:
            continue
        if i > 999:
            break
        print(i, end=" ")
    

n, t = [int(x) for x in input().split()]
arr = [int(x) for x in input().split()]

if t == 1:
    action1(n, arr)
elif t == 2:
    action2(arr)
elif t == 3:
    action3(n, arr)
elif t == 4:
    action4(n, arr)
else:
    action5(arr)
